HIP-0006.md

```diff
@@ -0,0 +1,170 @@
+
+```
+Number:  HIP-0006
+Title:   Trustless Live Auctions for Handshake
+Type:    Informational
+Status:  Final
+Authors: Matthew Zipkin (@pinheadmz)
+Created: 2021-06-17
+```
+
+## Abstract
+
+We propose an ascending-price auction system for HNS domain names without trust or
+third-party custody. A name owner initiates an auction by locking their TLD with
+a verifiable script. Bidders place bids by partially-signing `TRANSFER` transactions,
+and the name owner eventually picks the highest bid and atomically swaps the name
+ownership for the payment. The protocol requires more interactivity and centralization
+than the current Shakedex-style HIP-0001 auction format, but with some compelling trade-offs.
+
+## Definitions
+
+This proposal is based on [HIP-0001](https://hsd-dev.org/HIPs/proposals/0001/), and
+in particular the Swap Script, Transaction Structure, and Swap Proof. Handshake
+primitives like `OP_TYPE` and `SIGHASH_SINGLEREVERSE` are also required.
+
+Recall the HIP-0001 Swap Script only allows two covenant actions be applied to the name:
+
+- The name owner can sign a TRANSFER with their private key
+
+- Anyone can FINALIZE without a signature
+
+## Protocol Summary
+
+### Auction OPEN
+
+The auction technically begins when a name owner FINALIZEs a transfer of their name
+to a HIP-0001 locking script. The name owner must then publicly reveal the redeem script
+for the address that owns the name, so it can be verified. The locking script prevents
+the name owner from canceling an in-progress name ownership transfer. Since there is a
+small amount of interactivity required in the protocol, the name owner must also publish
+"where" the auction will take place. The auction "location" is just some internet endpoint
+where bidders can submit bids. It can be a private email address (in which case the name owner
+may choose to keep all bids confidential or truly blind), or it can be a public server
+that announces every new bid as they arrive.
+
+### Auction BID
+
+A user that wants to bid on the name constructs a partially-signed transaction in
+the following format:
+
+```
+vin:
+  0: current name owner, encumbered by Swap Script
+  1: A UTXO controlled by the bidder of some value V
+vout:
+  0: TRANSFER to the bidder's address
+```
+
+Input 1 is signed with `SIGHASH_ANYONECANPAY | SIGHASH_SINGLEREVERSE`. This commits
+the incoming payment to the name transfer. Note that Input 0 is not signed yet,
+but it must be the current name-owning UTXO or the linked output 0 would be invalid.
+
+Note that this transaction has no change output. Because the bidder is signing
+with single-reverse, they can't specify a change output in addition to the name
+transfer output. If they were to sign with `SIGHASH_ALL` then the next step of
+the protocol would not be possible. If they signed with `SIGHASH_SINGLE` they could
+commit to a change address, but not the name transfer.
+
+Having no change output means the INPUT coin they are spending MUST BE THE VALUE
```

Comment:
Is this caveat important enough to be explicitly mentioned again in a separate "Tradeoffs" or "Caveats" or something like that section, along with the "Seller pays the miner fee" below?

Reply:
I'll mention both of these tradeoffs in the abstract, thanks.